app/views/incomeSources/add/IncomeSourceCheckDetails.scala.html

```diff
@@ -0,0 +1,208 @@
+        }
+    }
+
+@mainTemplate(
+    pageTitle = getMessage("title"),
+    backUrl = Some(backUrl),
+    useFallbackBackLink = true,
+    isAgent = isAgent,
+    btaNavPartial = user.btaNavPartial) {
+
+    @hmrcPageHeading(PageHeading(
+        text = getMessage("title"),
+        headingClasses = Some("govuk-heading-l"),
+        section = Some(getMessage("heading")),
+        captionClasses = Some("govuk-caption-l")
+    ))
+
+    @formWithCSRF(action = postAction) {
+
+        <dl class="govuk-summary-list" id="check-details-table">
+            @if(viewModel.incomeSourceType == SelfEmployment){
+                <div class="govuk-summary-list__row">
+                    <dt class="govuk-summary-list__key">@getMessage("business-name")</dt>
+                    <dd class="govuk-summary-list__value" id="business-name-value">@{
+                        viewModel.businessName.getOrElse("")
+                    }</dd>
+                    <dd class="govuk-summary-list__actions govuk-!-text-align-right govuk-!-margin-0">
+                        <a class="govuk-link" href=@changeBusinessNameURL>
+                        @getMessage("change")
+                        </a>
+                    </dd>
+                </div>
+            }
+            <div class="govuk-summary-list__row">
+                <dt class="govuk-summary-list__key">@getMessage("start-date")</dt>
+                <dd class="govuk-summary-list__value" id="start-date-value">@{
+                    viewModel.businessStartDate.get.toLongDate
+                }</dd>
+                <dd class="govuk-summary-list__actions govuk-!-text-align-right govuk-!-margin-0">
+                    <a class="govuk-link" id="change-start-date-link" href=@changeStartDateURL>
+                    @getMessage("change")
+                    </a>
+                </dd>
+            </div>
+            @if(viewModel.incomeSourceType == SelfEmployment){
+                <div class="govuk-summary-list__row">
+                    <dt class="govuk-summary-list__key">@getMessage("business-description")</dt>
+                    <dd class="govuk-summary-list__value" id="business-trade-value">@{
+                        viewModel.businessTrade
+                    }</dd>
+                    <dd class="govuk-summary-list__actions govuk-!-text-align-right govuk-!-margin-0">
+                        <a class="govuk-link" href=@changeBusinessTradeURL>
+                        @getMessage("change")
+                        </a>
+                    </dd>
+                </div>
+                <div class="govuk-summary-list__row">
+                    <dt class="govuk-summary-list__key">@getMessage("business-address")</dt>
+                    <dd class="govuk-summary-list__value" id="business-address-value">@{
+                        HtmlFormat.raw(getLongAddressFromBusinessAddressDetails(
+                            address1 = viewModel.businessAddressLine1.getOrElse(""),
```

Comment:
as far as I remember businessAddressLine1 would be always provided so we can change this to be not Optional